answer/Explanation.java

```diff
@@ -32,14 +35,21 @@
 public class Explanation {
 
     private final List<ConceptMap> answers;
+    private final Rule rule;
 
-    public Explanation(List<ConceptMap> ans) {
+    public Explanation(List<ConceptMap> ans, @Nullable Rule rule) {
```

Comment:
Why can the `rule` be `@Nullable`?

Reply:
Because we have (on the server side) 3 kinds of explanation - JoinExplanation, LookupExplanation and RuleExplanation. Only the latter actually shows the derivation of facts from a rule. 
I'm inclined to think that these could be named `Resolution`s rather than `Explanation`s.

Reply:
👍